### PR TITLE
(WIP) CRM-15861: Offline membership renewal doesn't display priceset choices 

### PR DESCRIPTION
**Pull request by litespeedmarc**
This needs a good functional review & some QA testing.  I've done quite a fair bit here, but need someone else to wring it through and give me a hard time on it (but not too hard a time).

Also, some code review would be nice -- though keeping in mind, the membership stuff is pretty nasty and riddled with "I'm-only-going-to-cover-my-case" branches.  I've tried to keep these to a minimum, but being new, and without a serious overhaul (and the risks that would come with that), the changes are what they are.

See https://issues.civicrm.org/jira/browse/CRM-15861 for more details

### CODE REVIEW

**Comment by totten** [comment]
jenkins, add to whitelist

File: CRM/Member/BAO/Membership.php [hunk 1]
```diff
@@ -2508,4 +2514,93 @@ public static function getContactsCancelledMembership($contactID, $isTest = FALS
...
+                    case when mem.status_id in (select id from civicrm_membership_status where is_current_member)
+                         then 1
+                         else 0
+                    end,
+                    end_date DESC
+                limit 1) existing_membership_id
+            from civicrm_membership_type mt1
+           where id in (" . implode(", ", array_keys($membershipTypes) ) . ");
+    ";
+    $dao = CRM_Core_DAO::executeQuery($query);
```
Comment (seamuslee001) [inline CRM/Member/BAO/Membership.php]: Same would apply here. but this time you probably want string for membershipTypes

Comment (litespeedmarc) [inline CRM/Member/BAO/Membership.php]: Done for `$contactId`.  Left `$membershipTypes` unchanged, as to handle this through CiviCRM's binding framework would add complexity, would also be inconsistent with the few hundred other places that do similar in clauses, and to be handled properly might be better handled through a new type (e.g., Int[]), which would be out of scope for this.

**Comment by eileenmcnaughton** [comment]
Hey - do you want to rebase this to clean it up a bit. What I normally do is

``git pull --rebase origin master
git push -f myrepo master
``

Generally after that I would squash some commits by doing 

``
git fetch -f origin
git rebase -i origin/master
git push -f myrepo master
``

In the interactive screen you can mark some commits 's' to squash them into the one above them

**Comment by litespeedmarc** [comment]
Closing this, will create a new one.  I've obviously messed up my git commands.  It'll be easier to just start new & apply my changes from stash.

**Pull request closed without merging**